adminSiteClient/SuggestedChartRevisionApproverPage.tsx

```diff
@@ -0,0 +1,1367 @@
+                        other feedback.
+                    </p>
+                    <p className="text-danger">
+                        WARNING: This tool is new and may contain bugs that
+                        cause unexpected behavior. Use with caution.
+                    </p>
+                    {this.renderReadme()}
+                    {this.renderSettings()}
+
+                    {this.numTotalRows > 0 || !this.listMode ? (
+                        this.renderApprovalTool()
+                    ) : (
+                        <div style={{ paddingBottom: 20 }}>
+                            0 pending chart revisions found. All suggested chart
+                            revisions have already been approved or rejected.
+                        </div>
+                    )}
+                </main>
+            </AdminLayout>
+        )
+    }
+
+    renderApprovalTool() {
+        const status =
+            this.suggestedChartRevision && this.suggestedChartRevision.status
+        return (
+            <React.Fragment>
+                {this.renderControls()}
+                <h3>
+                    Suggested revision{" "}
+                    {this.suggestedChartRevision
+                        ? this.suggestedChartRevision.id
+                        : ""}
+                    <span
+                        className={classNames({
+                            "text-primary":
+                                status ===
+                                SuggestedChartRevisionStatus.approved,
+                            "text-danger":
+                                status ===
+                                SuggestedChartRevisionStatus.rejected,
+                            "text-warning":
+                                status === SuggestedChartRevisionStatus.flagged,
+                            "text-secondary":
+                                status === SuggestedChartRevisionStatus.pending,
+                        })}
+                        style={{ marginLeft: "20px" }}
+                    >
+                        <SuggestedChartRevisionStatusIcon status={status} />{" "}
+                        {this.suggestedChartRevision ? (
+                            <i>
+                                {status.charAt(0).toUpperCase() +
+                                    status.slice(1)}
+                            </i>
+                        ) : (
+                            ""
+                        )}
+                    </span>
+                </h3>
+                {this.renderGraphers()}
+                {this.renderMeta()}
+            </React.Fragment>
+        )
+    }
+
+    renderReadme() {
+        return (
+            <div className="collapsible">
+                <h3>
+                    README
+                    <button
+                        className="btn btn-outline-dark"
+                        type="button"
+                        onClick={() => (this.showReadme = !this.showReadme)}
```

Comment:
AFAIK when using the MobX library that we use for state management then all modification of state should happen in functions that are marked as action (as you are doing on named functions). Alternatively you can also wrap this event handler call with the action function - see https://mobx.js.org/actions.html#wrapping-functions-using-action . Note that I am relatively new to mobx as well and maybe this does not apply here for some reason

Reply:
Yes, we should be doing this, and the existing codebase is a bad example. But technically, actions I think actions only make a difference when multiple observable properties are edited.

Reply:
resolved in 6387f7e2d8a96149102d8b663ffcd1aa47e3b40d